### ISSUE TITLE
inconsistent subset_h5.H5D with non consecutive indexes

### ISSUE THREAD

**Issue by jeanmanguy** [hhoeflin/hdf5r#123]
Hi,

I am trying to subset rows from a table stored in a hdf5 file, the variable `pair_id` should match the row index (minus 1 because pair_id starts at 0 and not 1 in R). I pasted the reprex below to show the inconsistent results. 

``` r
library(hdf5r)
path_file_prob <- "path/to/file.h5"
file_prob <- H5File$new(path_file_prob, mode = "r")
pairs = file_prob[['/probabilities/pairs']]

# correct
pairs[c(1) + 1 ][c('pair_id', 'prob')]
#>   pair_id prob
#> 1       1  180

# correct
pairs[c(1, 3, 4) + 1 ][c('pair_id', 'prob')]
#>   pair_id prob
#> 1       1  180
#> 2       3  137
#> 3       4  137

# correct
pairs[c(1, 2) + 1 ][c('pair_id', 'prob')]
#>   pair_id prob
#> 1       1  180
#> 2       2  137

# wrong
pairs[c(1, 3) + 1 ][c('pair_id', 'prob')]
#>   pair_id prob
#> 1       0  211
#> 2       2  137

# wrong
pairs[c(1, 3, 5) + 1 ][c('pair_id', 'prob')]
#>   pair_id prob
#> 1       0  211
#> 2       2  137
#> 3       4  137

# correct
pairs[c(20, 50, 600, 999) + 1 ][c('pair_id', 'prob')]
#>   pair_id prob
#> 1      19   -5
#> 2      49   -7
#> 3     600   -9
#> 4     999   10

# wrong
pairs[c(20, 50) + 1 ][c('pair_id', 'prob')]
#>   pair_id prob
#> 1      20   -5
#> 2      50  -11
```

<sup>Created on 2019-04-18 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by hhoeflin** [comment]
Thank you for letting me know. Could you also add the hdf5-file that you are using (or the relevant part of it)?

Thanks

**Comment by jeanmanguy** [comment]
I extracted a portion of the file, tested and found the same behaviour: [file2b656d51d28d.h5.zip](https://github.com/hhoeflin/hdf5r/files/3093862/file2b656d51d28d.h5.zip)

**Comment by hhoeflin** [comment]
thanks - looking into it.

**Comment by hhoeflin** [comment]
submitted a fix - see PR #124. I tried your example and for me it fixes it. Let me know if it works for you.

Thanks for submitting the issue.

**Comment by jeanmanguy** [comment]
The fix is working.

Thank you
